src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -193,21 +196,35 @@ bool PerformAudioPassThruRequest::PrepareResponseParameters(
     mobile_apis::Result::eType& result_code, std::string& info) {
   LOG4CXX_AUTO_TRACE(logger_);
 
+  bool result = true;
   ResponseInfo ui_perform_info(result_ui_, HmiInterfaces::HMI_INTERFACE_UI);
   ResponseInfo tts_perform_info(result_tts_speak_,
                                 HmiInterfaces::HMI_INTERFACE_TTS);
-  const bool result =
-      PrepareResultForMobileResponse(ui_perform_info, tts_perform_info);
+
+  PrepareResultForMobileResponse(ui_perform_info, tts_perform_info);
 
   if (ui_perform_info.is_ok && tts_perform_info.is_unsupported_resource &&
       HmiInterfaces::STATE_AVAILABLE == tts_perform_info.interface_state) {
     result_code = mobile_apis::Result::WARNINGS;
     tts_info_ = "Unsupported phoneme type sent in a prompt";
-    info = MergeInfos(ui_perform_info, ui_info_, tts_perform_info, tts_info_);
-    return result;
+  } else if (IsResultCodeUnsupported(ui_perform_info, tts_perform_info)) {
+    result_code = mobile_apis::Result::UNSUPPORTED_RESOURCE;
+  } else if (IsAnyHMIComponentAborted(ui_perform_info, tts_perform_info)) {
+    result_code = mobile_apis::Result::ABORTED;
+    result = false;
+  } else {
+    result_code = PrepareAudioPassThruResultCodeForResponse(ui_perform_info,
+                                                            tts_perform_info);
+    if (!ui_perform_info.is_ok) {
+      result = false;
+    }
   }
-  result_code = PrepareResultCodeForResponse(ui_perform_info, tts_perform_info);
+
   info = MergeInfos(ui_perform_info, ui_info_, tts_perform_info, tts_info_);
+  if (audio_pass_thru_icon_exists_ == false) {
+    info = MergeInfos("Reference image(s) not found", info);
```

Comment:
The info variable in this case, using MergeInfos method, will be extended, not rewritten.